plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundReadWrite.java

```diff
@@ -18,254 +18,63 @@
-import org.slf4j.LoggerFactory;
+import org.apache.commons.httpclient.HttpMethod;
+import org.apache.commons.httpclient.util.HttpURLConnection;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import edu.berkeley.ground.exceptions.GroundException;
+import edu.berkeley.ground.plugins.hive.util.PluginUtil;
 
 public class GroundReadWrite {
 
-  private static final String GROUNDCONF = "ground.properties";
-
-  static final private Logger LOG = LoggerFactory.getLogger(GroundReadWrite.class.getName());
-
-  static final String GRAPHFACTORY_CLASS = "ground.graph.factory";
-
-  static final String NODEFACTORY_CLASS = "ground.node.factory";
-
-  static final String EDGEFACTORY_CLASS = "ground.edge.factory";
-
-  static final String NO_CACHE_CONF = "no.use.cache";
-  private DBClient dbClient;
-  private GraphFactory graphFactory;
-  private GraphVersionFactory graphVersionFactory;
-  private NodeVersionFactory nodeVersionFactory;
-  private EdgeVersionFactory edgeVersionFactory;
-  private String factoryType;
-
-  @VisibleForTesting
-  final static String TEST_CONN = "test_connection";
-
-  protected static final String DEFAULT_FACTORY = "neo4j";
-
-  private static Configuration staticConf = null;
-
-  private NodeFactory nodeFactory;
-
-  private EdgeFactory edgeFactory;
-
-  private StructureVersionFactory structureVersionFactory;
-
-  private StructureFactory structureFactory;
-
-  private String dbName;
-
-  private String userName;
-
-  private String password;
-
-  private String host;
-
-  public StructureVersionFactory getStructureVersionFactory() {
-    return structureVersionFactory;
-  }
+    public static final String NO_CACHE_CONF = "no_cache_conf";
+    private GroundReadWriteNodeResource groundReadWriteNodeResource;
+    private GroundReadWriteStructureResource groundReadWriteStructureResource;
+    private GroundReadWriteEdgeResource groundReadWriteEdgeResource;
 
-  public void setStructureVersionFactory(StructureVersionFactory svf) {
-    this.structureVersionFactory = svf;
-  }
-
-  /**
-   * @return the sf
-   */
-  public StructureFactory getStructureFactory() {
-    return structureFactory;
-  }
-
-  /**
-   * @param sf the sf to set
-   */
-  public void setStructureFactory(StructureFactory sf) {
-    this.structureFactory = sf;
-  }
-
-  private static ThreadLocal<GroundReadWrite> self = new ThreadLocal<GroundReadWrite>() {
-    @Override
-    protected GroundReadWrite initialValue() {
-      if (staticConf == null) {
-        throw new RuntimeException("Must set conf object before getting an instance");
-      }
-      try {
-        return new GroundReadWrite(staticConf);
-      } catch (GroundDBException e) {
-        LOG.error("create groundreadwrite failed {}", e.getMessage());
-      }
-      return null;
+    public GroundReadWrite() throws GroundException {
+        // initialize all ground related resources
+        setGroundReadWriteNodeResource(new GroundReadWriteNodeResource());
```

Comment:
I wonder whether it would be better to declare the corresponding variables `final` and use a direct assignment here. It seems that this should be the only way to instantiate these resources, so why would we allow mutations after creating an instance of `GroundReadWrite` ?